tmt/steps/provision/__init__.py

```diff
@@ -146,8 +146,12 @@ def go(self):
                         self.info('')
                 finally:
                     if isinstance(phase, ProvisionPlugin):
-                        if phase.guest():
-                            self._guests.append(phase.guest())
+                        guest = phase.guest()
+                        # To decide whether the guest should be appended to
+                        # list 'self._guests', 'guest.is_ready' should NOT be
+                        # used right here because dry mode is supported
```

Comment:
Thank you for this explanation - however we _do_ know we are running in dry mode right? Adding _not_ ready guest moves to problem to the pipeline (all push, execute ... need to be fixed to check for readiness (well, they should check for that either way but that will require much more work))

So how about `if guest.is_ready or self.opt('dry'):` ?